### ISSUE TITLE
Can't import to react project

### ISSUE THREAD

**Issue by paniczklos** [Conflux-Chain/js-conflux-sdk#83]
Hi , I'm having porblems with importing to a react project on codesandbox. I'm using the 1.0.0-alpha4 version.

The error is :

```
Super expression must either be null or a function
t.exports
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:1:3436
eval
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:17:142726
Object.289../BaseProvider
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:17:143970
i
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:1:655
eval
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:1:690
Object.290../BaseProvider
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:17:147013
i
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:1:655
eval
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:1:690
Object.252../Drip
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:17:68028
i
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:1:655
eval
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:1:690
Object.286../Conflux
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:17:134974
i
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:1:655
e
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:1:815
eval
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:1:832
eval
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:1:197
evaluate
https://2vk0q.csb.app/node_modules/js-conflux-sdk/dist/js-conflux-sdk.umd.min.js:1:383
z
https://codesandbox.io/static/js/sandbox.ac6a13f34.js:1:97548
Q.evaluate
https://codesandbox.io/static/js/sandbox.ac6a13f34.js:1:109662
me.evaluateTranspiledModule
https://codesandbox.io/static/js/sandbox.ac6a13f34.js:1:119233
c
https://codesandbox.io/static/js/sandbox.ac6a13f34.js:1:109408
evaluate
/src/components/AppRoot.js:7:20
   4 | import Col from "react-bootstrap/Col";
   5 | import "fontsource-roboto";
   6 | import { Route, NavLink, HashRouter } from "react-router-dom";
>  7 | const { Conflux } = require("js-conflux-sdk");
     |                    ^
   8 | 
   9 | 
  10 | import { gettronweb, getWeb3 } from "../services";
View compiled
▶ 4 stack frames were collapsed.
evaluate
/src/index.js:5
  2 | import ReactDOM from "react-dom";
  3 | import "bootstrap/dist/css/bootstrap.css";
  4 | import "./styles.css";
> 5 | import { AppRoot } from "./components/AppRoot";
  6 | 
  7 | ReactDOM.render(<AppRoot />, document.getElementById("root"));
  8 | 
View compiled
▼ 10 stack frames were expanded.
z
https://codesandbox.io/static/js/sandbox.ac6a13f34.js:1:97548
Q.evaluate
https://codesandbox.io/static/js/sandbox.ac6a13f34.js:1:109662
me.evaluateTranspiledModule
https://codesandbox.io/static/js/sandbox.ac6a13f34.js:1:119233
me.evaluateModule
https://codesandbox.io/static/js/sandbox.ac6a13f34.js:1:118969
(anonymous function)
https://codesandbox.io/static/js/sandbox.ac6a13f34.js:1:800938
l
https://codesandbox.io/static/js/common-sandbox.43541997b.chunk.js:1:256735
Generator._invoke
https://codesandbox.io/static/js/common-sandbox.43541997b.chunk.js:1:256488
Generator.forEach.e.<computed> [as next]
https://codesandbox.io/static/js/common-sandbox.43541997b.chunk.js:1:257092
t
https://codesandbox.io/static/js/common-sandbox.43541997b.chunk.js:1:2405
i
https://codesandbox.io/static/js/common-sandbox.43541997b.chunk.js:1:2616
```

**Comment by ShawkiS** [comment]
Hey @paniczklos, have you found a solution for it?

**Comment by boqiu** [comment]
@GeekBerry can you please fix this issue with priority?

**Comment by yqrashawn** [comment]
@paniczklos Hi, do you have the codesandbox url?

**Comment by yqrashawn** [comment]
@paniczklos can you try use `import` instead of `require` to introduce `js-conflux-sdk`

![image](https://user-images.githubusercontent.com/15090582/95809487-80cebd80-0d41-11eb-8e0b-e009fee9e927.png)

**Comment by Pana** [comment]
@paniczklos @ShawkiS You can use version 0.13, the current 1.0 is a alpha version, will release a new one when mainnet online

**Comment by GeekBerry** [comment]
is js-conflux-sdk@1.0.0 resolve your problem?

**Comment by aalu1418** [comment]
@GeekBerry @Pana i just tried the latest release of the js-conflux-sdk and it still doesn't work.

Details:
Used `create-react-app` to get setup
Dependencies:
```
    "@testing-library/jest-dom": "^5.11.4",
    "@testing-library/react": "^11.1.0",
    "@testing-library/user-event": "^12.1.10",
    "js-conflux-sdk": "^1.0.1",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-scripts": "4.0.0",
    "web-vitals": "^0.2.4"
```
Here's the sample code that i'm working with: https://github.com/Conflux-Network-Global/webinar-code/tree/dapp-refactor/sample-dapp

**Comment by aalu1418** [comment]
I did some checking in the Javascript SDK, the error happens in the `WebSocketProvider.js` file. it seems like the `websocket.client` is `undefined` which is causing the `super(options)` line to fail?

It might be good to disable the websocket functionality for now since it seems like it's incomplete? It doesn't affect operation when running the package only as a node.js file, but it breaks when using the package with React

**Comment by GeekBerry** [comment]
em..., it is a problem, and do you have any idea about:
* is [websocket](https://www.npmjs.com/package/websocket) usable for react ?
* how to use websocket in react
* is there any way to check and disable websocket when environment is react, and did web3js has the same problem?

**Comment by Pana** [comment]
We have found a solution for this, we are dealing this

**Comment by GeekBerry** [comment]
i think we fix it by using W3C native websocket in `WebsocketProvider` and published as [v1.0.2](https://www.npmjs.com/package/js-conflux-sdk/v/1.0.2),
thank your for your report.

**Comment by aalu1418** [comment]
@GeekBerry thank you! it seems like it is working now. I tested it in here: https://github.com/Conflux-Network-Global/webinar-code/tree/master/sample-dapp
